PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx

```diff
@@ -0,0 +1,187 @@
+  ConfigurableAxis axisPtAssoc{"axisPtAssoc", {VARIABLE_WIDTH, 0.5, 1.0, 1.5, 2.0, 3.0, 4.0, 6.0, 10.0}, "pt associated axis for histograms"};
+  ConfigurableAxis axisMultiplicity{"axisMultiplicity", {VARIABLE_WIDTH, 0, 5, 10, 15, 20, 25, 30, 35, 40, 50, 60, 80, 100}, "multiplicity / centrality axis for histograms"};
+
+  HistogramRegistry registry{"registry"};
+  int logcolls = 0;
+  int logcollpairs = 0;
+
+  void init(InitContext&)
+  {
+    LOGF(info, "Starting init");
+    registry.add("Yield", "pT vs eta vs Nch", {HistType::kTH3F, {{40, 0, 20, "p_{T}"}, {100, -2, 2, "#eta"}, {100, 0, 100, "Nch"}}}); // check to see total number of tracks
+    registry.add("etaphi_Trigger", "eta vs phi vs Nch", {HistType::kTH3F, {{100, -2, 2, "#eta"}, {200, 0, 2 * M_PI, "#varphi"}, {100, 0, 100, "Nch"}}});
+
+    registry.add("deltaEta_deltaPhi_same", "", {HistType::kTH2D, {axisDeltaPhi, axisDeltaEta}}); // check to see the delta eta and delta phi distribution
+    registry.add("deltaEta_deltaPhi_mixed", "", {HistType::kTH2D, {axisDeltaPhi, axisDeltaEta}});
+
+    registry.add("Sparse_mixed", "", {HistType::kTHnSparseD, {{axisVertex, axisPtTrigger, axisPtAssoc, axisMultiplicity, axisDeltaPhi, axisDeltaEta}}}); // Make the output sparse
+    registry.add("Sparse_same", "", {HistType::kTHnSparseD, {{axisVertex, axisPtTrigger, axisPtAssoc, axisMultiplicity, axisDeltaPhi, axisDeltaEta}}});
+
+    const int maxMixBin = axisMultiplicity->size() * axisVertex->size();
+    registry.add("eventcount", "bin", {HistType::kTH1F, {{maxMixBin + 2, -2.5, -0.5 + maxMixBin, "bin"}}}); // histogram to see how many events are in the same and mixed event
+  }
+
+  template <typename TCollision, typename TTracks>
+  void fillYield(TCollision collision, float centrality, TTracks tracks) // function to fill the yield and etaphi histograms. (This is not needed can be removed)
+  {
+    for (auto& track1 : tracks) {
+      registry.fill(HIST("Yield"), track1.pt(), track1.eta(), track1.size());
+      registry.fill(HIST("etaphi_Trigger"), track1.size(), track1.phi(), track1.eta());
+    }
+  }
+
+  template <typename TCollision>
+  bool fillCollision(TCollision collision, float centrality)
+  {
+
+    if (!collision.sel8()) {
+      return false;
+    }
+
+    return true;
+  }
+
+  template <typename TTracks>
+  void fillCorrelations(TTracks tracks1, TTracks tracks2, float posZ, int system, float Nch) // function to fill the Output functions (sparse) and the delta eta and delta phi histograms
+  {
+    // loop over all tracks
+    for (auto& track1 : tracks1) {
+
+      for (auto& track2 : tracks2) {
+        if (track1 == track2) {
+          continue;
+        }
+
+        float deltaPhi = track1.phi() - track2.phi();
+        float deltaEta = track1.eta() - track2.eta();
+        if (deltaPhi > 1.5f * PI) {
+          deltaPhi -= TwoPI;
+        }
+        if (deltaPhi < -PIHalf) {
+          deltaPhi += TwoPI;
+        }
```

Comment:
Do not duplicate already available utilities. See `RecoDecay::constrainAngle`.

Reply:
You can find it on https://github.com/AliceO2Group/O2Physics/blob/master/Common/Core/RecoDecay.h